File: api/create-checkout.js
const fs = require('fs');
const path = require('path');
const Stripe = require('stripe');

module.exports = async (req, res) => {
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  const stripeKey = process.env.STRIPE_SECRET_KEY;
  if (!stripeKey) {
    return res.status(500).json({ error: 'Stripe is not configured on this deployment yet' });
  }

  const stripe = new Stripe(stripeKey);
  const body = typeof req.body === 'string' ? JSON.parse(req.body || '{}') : (req.body || {});

  if (!body.items || !Array.isArray(body.items) || !body.items.length) {
    return res.status(400).json({ error: 'No items provided' });
  }

  const productsPath = path.join(process.cwd(), 'products.json');
  const products = JSON.parse(fs.readFileSync(productsPath, 'utf8'));
  const byId = Object.fromEntries(products.map(p => [p.id, p]));

  const line_items = [];
  for (const item of body.items) {
    const product = byId[item.id];
    if (!product || product.active === false) {
      return res.status(400).json({ error: `Unknown product: ${item.id}` });
    }
    const qty = Math.max(1, parseInt(item.qty || 1, 10));
    line_items.push({
      price_data: {
        currency: 'usd',
        unit_amount: Math.round(Number(product.price) * 100),
        product_data: { name: product.name }
      },
      quantity: qty
    });
  }

  try {
    const origin = req.headers.origin || 'https://storexpress-nine.vercel.app';
    const session = await stripe.checkout.sessions.create({
      payment_method_types: ['card'],
      line_items,
      mode: 'payment',
      success_url: `${origin}/cart.html?success=1`,
      cancel_url: `${origin}/cart.html?canceled=1`
    });
    return res.status(200).json({ url: session.url });
  } catch (error) {
    return res.status(500).json({ error: error.message || 'Stripe checkout failed' });
  }
};

File: api/create-checkout.py
import json
import os
from pathlib import Path

import stripe

PRODUCTS_FILE = Path(__file__).resolve().parent.parent / 'products.json'
STRIPE_SECRET_KEY = os.environ.get('STRIPE_SECRET_KEY')
stripe.api_key = STRIPE_SECRET_KEY if STRIPE_SECRET_KEY else ''


def load_products():
    with PRODUCTS_FILE.open() as f:
        return {p['id']: p for p in json.load(f)}


def handler(request):
    if request.method != 'POST':
        return {
            'statusCode': 405,
            'headers': {'Content-Type': 'application/json'},
            'body': json.dumps({'error': 'Method not allowed'})
        }

    try:
        body = request.get_json() if hasattr(request, 'get_json') else json.loads(request.body)
    except Exception:
        body = None

    if not body or not body.get('items'):
        return {
            'statusCode': 400,
            'headers': {'Content-Type': 'application/json'},
            'body': json.dumps({'error': 'No items provided'})
        }

    if not STRIPE_SECRET_KEY:
        return {
            'statusCode': 500,
            'headers': {'Content-Type': 'application/json'},
            'body': json.dumps({'error': 'Stripe is not configured on this deployment yet'})
        }

    products = load_products()
    line_items = []

    for item in body['items']:
        product = products.get(item.get('id'))
        if not product or product.get('active') is False:
            return {
                'statusCode': 400,
                'headers': {'Content-Type': 'application/json'},
                'body': json.dumps({'error': f"Unknown product: {item.get('id')}"})
            }
        qty = max(1, int(item.get('qty', 1)))
        line_items.append({
            'price_data': {
                'currency': 'usd',
                'unit_amount': int(float(product['price']) * 100),
                'product_data': {'name': product['name']}
            },
            'quantity': qty
        })

    origin = request.headers.get('origin') or request.headers.get('Origin') or 'https://storexpress.vercel.app'

    try:
        session = stripe.checkout.Session.create(
            payment_method_types=['card'],
            line_items=line_items,
            mode='payment',
            success_url=f'{origin}/cart.html?success=1',
            cancel_url=f'{origin}/cart.html?canceled=1'
        )
    except Exception as e:
        return {
            'statusCode': 500,
            'headers': {'Content-Type': 'application/json'},
            'body': json.dumps({'error': str(e)})
        }

    return {
        'statusCode': 200,
        'headers': {'Content-Type': 'application/json'},
        'body': json.dumps({'url': session.url})
    }
